### ISSUE TITLE
the focus shifts from the list of notebooks to the notes within a few seconds

### ISSUE THREAD

**Issue by ericdq** [laurent22/joplin#15029]
### Operating system

Windows

### Joplin version

3.6.7

### Desktop version info

Dispositivo: win32, Intel(R) Core(TM) i3-10100Y CPU @ 1.30GHz
ID del Cliente: 1f3a5d1f4f5a4f67968404307cf6d35b
Versión de la Sincronización: 3
Versión del Perfil: 49
Llavero Soportado: Sí
ID de instancia alternativa: -
Sync target: Servidor de Joplin
Editor: Markdown

Revisión: af0cab6

Backup: 1.5.1
Freehand Drawing: 4.3.0

### Current behaviour

When I’m in the notebook list and move the cursor over a notebook that contains sub-notebooks (regardless of whether it’s expanded or collapsed), after a couple of seconds the focus moves to the corresponding notes list. Currently, it does not meet the WCAG 3.2.2 criteria.

Something similar also happens when I jump to a note using the ‘Go To’ option. The focus sometimes remains on the list of notes instead of focusing on the note I pressed Enter on.

These behaviours have been occurring for the last two or three versions. I couldn’t say exactly when they started. I’ve noticed that some changes were made that could affect these behaviours.

### Expected behaviour

I would expect the focus to remain on the notebook I am currently on in the notebook list, whether I navigate using the arrow keys or the keyboard letters. At most, an expanded notebook should collapse when I switch to another level. The focus should remain there until I press Enter to move to the corresponding notes list.

In the case of ‘Go to’, it should jump directly to the selected note when I press Enter. Currently, the behaviour is inconsistent. Sometimes it jumps directly, and other times it forces the user to press Enter twice: once to confirm the selection from the list, and again to jump from the list of notes to the note itself.

I seem to recall that there used to be an option to jump to the body of the note or to the title. In recent versions, this option has been lost and I always end up at the title.

I would like to request a return to the previous behaviour, which focused on both situations. In the case of the list of notebooks, I would at least like it to remain visible even if the different levels collapse when moving to the next one.

**Comment by personalizedrefrigerator** [comment]
> When I’m in the notebook list and move the cursor over a notebook that contains sub-notebooks (regardless of whether it’s expanded or collapsed), after a couple of seconds the focus moves to the corresponding notes list. Currently, it does not meet the WCAG 3.2.2 criteria.

Note: I can reproduce this on Fedora 43 Linux (GNOME) (this isn't Windows-specific).

**Comment by personalizedrefrigerator** [comment]
Based on a [`git bisect`](https://git-scm.com/docs/git-bisect), this seems to be a regression from https://github.com/laurent22/joplin/pull/14650. (Note: Bisecting was done based on the notebook list focus jump issue).

**Comment by ericdq** [comment]
Thank you for looking into the issue, @personalizedrefrigerator  From the description of the updates, I wasn’t sure which one might have caused this behaviour.

**Comment by personalizedrefrigerator** [comment]
https://github.com/laurent22/joplin/pull/14650 also seems to cause focus to jump to the note list if notes are removed from the current folder by a background task (e.g. sync).

**Comment by laurent22** [comment]
@personalizedrefrigerator, would it make sense to simply revert the commit that caused the regression? I see the open PR to fix this is complex and I wouldn't be surprised if it also introduces new issues, but please advise on what would be best since you looked into it

**Comment by personalizedrefrigerator** [comment]
> I wouldn't be surprised if it also introduces new issues

The main change in the linked pull request seems to be adding an "is the note list focused" check before auto-focusing (and changes are localized to `useRefocusOnDeletion` and its tests), so I think it's unlikely to cause regressions.

> I see the open PR to fix this is complex and I wouldn't be surprised if it also introduces new issues, but please advise on what would be best since you looked into it

This is a significant enough issue that it could make sense to revert https://github.com/laurent22/joplin/pull/14650.

Additionally, if I revert #14650 locally, I don't seem to be able to reproduce https://github.com/laurent22/joplin/issues/10753. It's possible that https://github.com/laurent22/joplin/issues/10753 was fixed before #14650.

**Comment by justin212407** [comment]
Hey @personalizedrefrigerator I saw that you have opened a subsequent pr for the issue. Should i close mine to keep the pr list manageable?
